### PR TITLE
ui

### LINKED ISSUE #19: sprites

### LINKED ISSUE #21: in game hud

### PR DESCRIPTION
**Pull request by anotherLostKitten**

### CODE REVIEW

File: src/client_gameloop.rs [hunk 1]
```diff
@@ -82,7 +81,7 @@ pub fn gameloop() -> Result<(), String> {
         std::thread::sleep(Duration::new(0, 1_000_000_000u32 / FRAMERATE));
     }
 
-    gs.handle.join().unwrap().unwrap();
+    (move || {gs.handle.join().unwrap().unwrap();})();
```
Comment (jesings) [inline src/client_gameloop.rs]: This closure is probably unnecessary at this point

File: src/map/grid.rs [hunk 2]
```diff
@@ -4,7 +4,7 @@ pub const MAPDIM: (usize, usize) = (640, 480);
 
 #[derive(Debug)]
 pub struct Tile {
-    pub texture: u8, //Not sure texture should be a u8, we can make it an SDL object later
+    pub texture: i32, //Not sure texture should be a u8, we can make it an SDL object later
```
Comment (jesings) [inline src/map/grid.rs]: Bad comment now

File: src/render/render.rs [hunk 3]
```diff
@@ -18,54 +20,58 @@ impl gamestate::ClientGamestate {
         let pos = gd.players[pid].lock().expect("Could not lock player to get its position.").pos().clone();
 
         let mut canv = self.sdl.canv.lock().expect("Could not lock canvas for rendering!");
-        //canv.clear(); unneeded
+        canv.set_draw_color(Color::RGB(0, 0, 0));
+        canv.clear();
+        let textures = &self.sdl.texture_table;
```
Comment (jesings) [inline src/render/render.rs:27]: Yum!

File: src/render/render.rs [hunk 4]
```diff
@@ -18,54 +20,58 @@ impl gamestate::ClientGamestate {
...
                 //now we have the position of the tile from its location relative to the player
-                
 
                 if col_index >= 0 && row_index >= 0 && col_index < gd.grid.cols as i32 && row_index < gd.grid.rows as i32 {
                     let grid_index = row_index as usize * gd.grid.cols + col_index as usize;
                     let tile = &gd.grid.tiles[grid_index];
-                    let rshade = tile.texture;
 
-                    canv.set_draw_color(Color::RGB(rshade, rshade, rshade));
+                    //canv.set_draw_color(Color::RGB(rshade, rshade, rshade));
```
Comment (jesings) [inline src/render/render.rs]: Get rid of commented line

File: src/render/texture_table.rs [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+pub struct TextureTable<'a> {
+    tiles_texture: Texture<'a>,
+    player_texture: Texture<'a>,
+    portrait_texture: Texture<'a>,
+}
+
+impl<'a> TextureTable<'a> {
+    pub fn init(texture_creator: &'a TextureCreator<sdl2::video::WindowContext>) -> Self {
+        let mut tiles_texture = texture_creator.load_texture(Path::new("textures/tiles_placeholder.png")).unwrap();
+        tiles_texture.set_blend_mode(BlendMode::Blend);
```
Comment (jesings) [inline src/render/texture_table.rs]: I think this can be removed

**Review by jesings — APPROVED** [review]
Turkce Cunguz

File: src/render/texture_table.rs (hunk 5 above)
Comment (anotherLostKitten) [reply src/render/texture_table.rs]: yes it could be removed gOOD point

File: src/render/render.rs (hunk 4 above)
Comment (anotherLostKitten) [reply src/render/render.rs]: yes it could be removed gOOD point

File: src/map/grid.rs (hunk 2 above)
Comment (anotherLostKitten) [reply src/map/grid.rs]: yes it could be removed gOOD point

File: src/client_gameloop.rs (hunk 1 above)
Comment (anotherLostKitten) [reply src/client_gameloop.rs]: yes it could be removed gOOD point

**Review by jesings — APPROVED** [review]
The Kingdom of Serbs and Croats but miss me with them Slovenes

**Pull request merged**